File: Product_APP_MVC/app/src/main/java/com/example/product_app_mvc/DB/ProductsLocalDataSource.java
package com.example.product_app_mvc.DB;

import androidx.lifecycle.LiveData;

import com.example.product_app_mvc.model.POJO_class;

import java.util.List;

public interface ProductsLocalDataSource {



    public LiveData<List<POJO_class>> getStoredData();

    public LiveData<Integer> getProductCount();

    public void delete(POJO_class Product);

    public void insert(POJO_class Product);
}

File: Android_Development/05_Android_Kotlin/Myapp/MyApplication/app/src/main/java/com/example/myapplication/database/ProductsLocalDataSource.kt
package com.example.myapplication.database

import androidx.lifecycle.LiveData
import com.example.myapplication.model.Product
import kotlinx.coroutines.flow.Flow

interface ProductsLocalDataSource {
    suspend fun addProduct(product: Product)
    suspend fun removeProduct(product: Product)
    fun getFavoriteProducts(): Flow<List<Product>>
    suspend fun getProductById(id: Int): Product?
}
